orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/TaskNode.java

```diff
@@ -24,26 +24,40 @@
 import com.google.common.annotations.VisibleForTesting;
 import com.netflix.spinnaker.orca.ExecutionStatus;
 import lombok.Value;
+import static com.netflix.spinnaker.orca.pipeline.TaskNode.GraphType.loop;
 
 /**
  * A node in a {@link TaskGraph} which can be either an individual task or a
  * sub-graph.
  */
 public interface TaskNode {
 
-  static TaskGraph build(Consumer<Builder> closure) {
-    Builder builder = new Builder();
+  static enum GraphType {
+    full, loop, head, tail
```

Comment:
lowercase enum threw me for a loop when reading this ... I'd suggest capitalized as they're constants.